server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -618,6 +869,60 @@ ClusterState openIndices(final Index[] indices, final ClusterState currentState)
         return ClusterState.builder(updatedState).routingTable(routingTable.build()).build();
     }
 
+    static Tuple<ClusterState, Collection<AddBlockResult>> finalizeBlock(final ClusterState currentState,
```

Comment:
Javadocs please :) they really do help when code spelunking in the future (at least I find them incredibly useful)